numpy/core/src/umath/ufunc_object.c

```diff
@@ -1983,7 +2107,7 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
     int nin, nout;
     int i, j, idim, nop;
     const char *ufunc_name;
-    int retval = -1, subok = 1;
+    int retval = 0, subok = 1;
```

Comment:
True, but this the correct base case if we end up with a code path that never sets `retval`, so seems like a good thing to keep